### PR TITLE
Add distributionSince and varianceSince methods to persistence extensions

### PR DESCRIPTION
**Pull request by edog1973**
Added methods to the persistence extension for distributionSince and varianceSince that allow finding the variance and standard distribution of a value over a given time.

### CODE REVIEW

File: bundles/core/org.openhab.core.persistence/src/main/java/org/openhab/core/persistence/extensions/PersistenceExtensions.java [hunk 1]
```diff
@@ -393,6 +393,90 @@ static public DecimalType averageSince(Item item, AbstractInstant timestamp, Str
...
+	 * Gets the standard deviation value of the state of a given <code>item</code> since a certain point in time. 
+	 * The {@link PersistenceService} identified by the <code>serviceName</code> is used. 
+	 * 
+	 * @param item the item to get the average state value for
+	 * @param the point in time to start the check 
+	 * @param serviceName the name of the {@link PersistenceService} to use
+	 * @return the standard deviation of the value since the given point in time
+	 */
+	static public DecimalType deviationSince(Item item, AbstractInstant timestamp, String serviceName) {
+		DecimalType variance = varianceSince(item, timestamp, defaultService);
```
Comment (teichsta) [inline bundles/core/org.openhab.core.persistence/src/main/java/org/openhab/core/persistence/extensions/PersistenceExtensions.java]: please change "defaultService" into "serviceName" here

**Comment by edog1973** [comment]
After doing more testing I noticed an issue.  When using data that has the current value already persisted to the database the calculations were wrong because the current value was included twice  To mitigate this the current value is now only included in the calculation if it's value is the same as the newest db record.  The issue affected the averageSince so I updated it's logic as well.

**Comment by teichsta** [comment]
looks good to me. Could you please squash your commits into one? Thanks, Thomas E.-E.

**Comment by kaikreuzer** [comment]
Does this apply to ESH as well?

**Comment by teichsta** [comment]
@edog1973 could you finally please squash your commits into one? Thanks, Thomas E.-E.

**Comment by teichsta** [comment]
@edog1973 any chance to squash now? Will reschedule to 1.8 otherwise …

**Pull request closed without merging**